### DISCUSSION TITLE
Add config to explicitly set 'usingTypeScript'

### DISCUSSION THREAD

**Discussion by olsonpm** [vercel/next.js#74216] [Ideas]
### Goals

I'm building an editor and have a file "initial-code.ts" which I'm importing as text, not as typescript.  I want to keep its .ts extension for syntax highlighting when I edit it.

during `next build` this file causes next to assume my project uses typescript - breaking when an imported dependency in that file doesn't exist in my project.

by adding a config 'usingTypeScript' I can tell next to skip unwanted typescript processes

### Background

I didn't see any discourse when checking github discussions and issues for the initial error I received "Linting and checking validity of types".  Also googled for a while to learn what was going on.

I looked at next.js's code and didn't see a sensible way to get around its assumption that my project used typescript.  I understand wanting to keep things easy for folk and auto-detecting TS - but believe a config should exist to be explicit.

### Proposal

[Here is my initial implementation](https://github.com/olsonpm/next.js/compare/695a585405c20130ce205d73aa9749bcd48550da...olsonpm:next.js:f2af587391341a9c3cc537a868d305175e9d9d13?w=1)

it adds a new config 'usingTypeScript' which defaults to 'auto' preserving existing behavior.  By setting it to `true` or `false`, it skips the intent step and responds appropriately.

**Comment by Netail** [comment]
Why not respect file extensions? Like using the `.txt` extension instead.
Perhaps you can exclude the file in the tsconfig? (if that works)

**Reply by olsonpm** [reply]
> I want to keep its .ts extension for syntax highlighting when I edit it.

and tsconfig opts next.js into its typescript processes as well  which is exactly what i want to avoid.
